esmvalcore/preprocessor/_mask.py

```diff
@@ -52,11 +52,11 @@ def _get_fx_mask(fx_data, fx_option, mask_type):
 def _apply_fx_mask(fx_mask, var_data):
     """Apply the fx data extracted mask on the actual processed data."""
     # Apply mask across
-    if np.ma.is_masked(var_data):
-        fx_mask |= var_data.mask
-
-    # Build the new masked data
-    var_data = np.ma.array(var_data, mask=fx_mask, fill_value=1e+20)
+    # Then apply the mask (with getmaskarray)
+    old_mask = da.ma.getmaskarray(var_data)
+    mask = old_mask | fx_mask
+    var_data = da.ma.masked_array(var_data, mask=mask)
```

Comment:
Ok ... Shall I create a new pull request?

Reply:
Are you sure about that V? We create masked arrays in many parts of our code, and this is the only instance where the `fill_value` is explicitly set:

```bash
> grep -Irn --include="*.py" "ma.masked_array"
esmvalcore/cmor/_fixes/shared.py:61:    # of `np.ma.filled` and `np.ma.masked_array`.
esmvalcore/cmor/_fixes/shared.py:78:    return num_module.ma.masked_array(array, mask=mask)
esmvalcore/preprocessor/_derive/vegfrac.py:43:        sftlf_cube.data = da.ma.masked_array(
esmvalcore/preprocessor/_mapping.py:136:    return np.ma.masked_array(data, mask)
esmvalcore/preprocessor/_mask.py:321:            cube.data = np.ma.masked_array(cube.data, mask)
esmvalcore/preprocessor/_mask.py:498:        cube.data = da.ma.masked_array(cube.core_data(), mask=mask)
esmvalcore/preprocessor/_mask.py:519:            cube.data = da.ma.masked_array(cube.core_data(), mask=mask)
...
```

We also do [set a fill value when saving the netcdf files](https://github.com/ESMValGroup/ESMValCore/blob/21ec6c22aba22b0a077d777c137def71488cd99f/esmvalcore/preprocessor/_io.py#L472). I am also not sure if we need this, but I guess it ensures that the fill value in the final netcdf is correct. We probably don't really care about the fill values in the arrays...

@joergbenke you can just commit and push to this branch, this will then show up in this pull request. No need to create a new one :+1:

Reply:
Let's fix this if we find an issue, the fill value is specified as save time and masked values will then be replaced by it.